graylog2-server/src/main/java/org/graylog2/outputs/IndexSetAwareMessageOutputBuffer.java

```diff
@@ -81,12 +101,15 @@ public void appendAndFlush(FilteredMessage filteredMessage, Consumer<List<Filter
         synchronized (this) {
             // See class the class documentation for the reasoning behind the bufferLength calculation.
             buffer.add(filteredMessage);
-            bufferLength.addAndGet(Math.max(filteredMessage.message().getIndexSets().size(), 1));
+            bufferLength += Math.max(filteredMessage.message().getIndexSets().size(), 1);
+            bufferSizeBytes += estimateOsBulkRequestSize(filteredMessage.message(), objectMapper);
```

Comment:
What do you think about skipping the serialization if `output_batch_size` is configured for a number of messages instead of bytes? Otherwise, we would always serialize early and have the memory overhead, even if we don't need it.